### PR TITLE
docs: Add testing composables section

### LINKED ISSUE #713: [Docs] Test usage of provide
It took me quite some time to get an example of how to test a `provide` of a component. I want to test if the component uses the `provide` function correctly and if it does some stuff with the injected data from possible child components. Which is somehow the opposite of the existing option which is already in-build to `vue-test-utils`:

```ts
mount(ChildComponent, {
  global: {  
    provide: { 
      title: 'My title'
    }
  }
})
```

I finally got it working with the following code. Maybe this helps others and can eventually be added to the docs.

I had some interesting errors on my way:
- adding a `name` to the childComponent breaks it
- leaving the `h(childComponent)` away breaks the slot (inside child: `getCurrentInstance().parent` is `null`)

## Sample

```vue
<template>
  <div class="parent">
    <p class="title">{{ title }}</p>
    <slot />
  </div>
</template>

 <script lang="ts">
import {defineComponent} from 'vue';

export defineComponent({
    setup() {
      const title = ref('Parent');
      provide('title', title);
      return { title };
    },
});
</script>
```

```ts
import { mount } from '@vue/test-utils';
import { h, inject, nextTick, provide, Ref, ref } from 'vue';
import parentComponent from './parentComponent';

describe('Repro', () => {
  it('should update injected title', async () => {
    expect.assertions(1);
    // given

    // the child component works like a mock to simulate missing inject option 
    const childComponent = {
      template: '<div />',

      setup() {
        const title = inject<Ref<string>>('title');
        if (title) {
          title.value = 'Child';
        }
        return {};
      },
    };

    // when
    const wrapper = mount(parentComponent, {
      slots: {
        default: h(childComponent),
      },
    });
    await nextTick();

    // then
    expect(wrapper.find('.title').text()).toStrictEqual(`Child`);
  });
});
```

### PR DESCRIPTION
**Pull request by freakzlike**
fix #713

### CODE REVIEW

**Comment by cexbrayat** [comment]
@freakzlike Let me know when this draft PR is ready to be reviewed 👍

**Comment by freakzlike** [comment]
@cexbrayat Please take a look. I'm not really good at writing documentions, but I hope it will help someone

File: docs/guide/advanced/reusability-composition.md [hunk 1]
```diff
@@ -2,6 +2,214 @@
...
+
+  const wrapper = mount(TestComponent, {
+    props: {
+      userId: 1
+    }
+  })
+
+  expect(wrapper.vm.user).toBeUndefined()
+
+  await flushPromises()
```
Comment (cexbrayat) [inline docs/guide/advanced/reusability-composition.md:87]: I suppose here you also need to mock the HTTP call one way or another.
IT's fine by me if we don't show how, but maybe a comment would be nice so readers don't expect this to work magically?

**Review by cexbrayat — APPROVED** [review]
Looks great! One minor comment, and I'm OK to merge 👍

File: docs/guide/advanced/reusability-composition.md (hunk 1 above)
Comment (freakzlike) [reply docs/guide/advanced/reusability-composition.md:90]: I added mocking the same way like in https://test-utils.vuejs.org/guide/advanced/async-suspense.html#resolving-other-asynchronous-behavior

**Pull request merged**